packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -348,8 +346,18 @@ class TreeViewExtImpl<T> implements Disposable {
         return idLabel;
     }
 
-    private buildTreeItemId(parentId: string, index: number, idLabel: string | undefined): string {
-        return `${parentId}/${index}:${idLabel}`;
+    private buildTreeItemId(parentId: string, item: TreeItem): string {
+        // build tree id according to https://code.visualstudio.com/api/references/vscode-api#TreeItem
+        // note: the front end tree implementation cannot handle reparenting items, hence the id is set to the "path" of individual ids
+        let id = typeof item.id === 'string' ? item.id : this.getItemLabel(item);
+        if (id) {
+            // we use '' as the id of the root, we don't consider that a valid id
+            // since '' is falsy, we'll never get '' in this branch
+            id = TreeViewExtImpl.ID_ITEM + id;
+        } else {
+            id = TreeViewExtImpl.ID_COMPUTED + this.nextItemId++;
+        }
+        return `${parentId}/${id}`;
```

Comment:
@tsmaeder, this turns out to be a dangerous way to build the ID. If `getItemLabel` returns the same thing for several children of the same item - we've seen this downstream in the references view when the lines where something is referred to are identical - then this produces children with the exact same ID's, which isn't permitted in our core `Tree` implementation. Should be able to reproduce in Theia in a moment.